controllers/org_rbac_controller.go

```diff
@@ -0,0 +1,121 @@
+package controllers
+
+import (
+	"context"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// OrganizationRBACReconciler reconciles RBAC rules for organization namespaces
+type OrganizationRBACReconciler struct {
+	client.Client
+	Recorder record.EventRecorder
+	Scheme   *runtime.Scheme
+
+	// OrganizationLabel is the label that marks to what organization (if any) the namespace belongs to
+	OrganizationLabel string
+	// DefaultClusterRoles is a map where the keys are the name of default rolebindings to create and the values are the names of the clusterroles they bind to
+	DefaultClusterRoles map[string]string
+}
+
+// LabelRoleBindingUninitialized is used to mark rolebindings as uninitialized.
+// In that case the controller will update it to bind to the organization.
+const LabelRoleBindingUninitialized = "appuio.io/uninitialized"
+
+//+kubebuilder:rbac:groups="",resources=namespaces,verbs=get;list;watch
+//+kubebuilder:rbac:groups="rbac.authorization.k8s.io",resources=rolebindings,verbs=get;list;watch;create;patch;update
+//+kubebuilder:rbac:groups="",resources=events,verbs=create;patch
+
+// Reconcile reacts to pod updates and emits events if the fair use request ratio is violated
```

Comment:
This should probably be updated to not be copy-pasted from the ratio controller :)

Reply:
:facepalm: